File: backend/src/main/java/model/Task.java
package model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import lombok.*;
import org.springframework.data.annotation.Id;

import java.util.Date;

@Entity
@Getter
@Setter
@RequiredArgsConstructor
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private final String title;
    private final String description;
    private final Date deadline;
}

File: frontend/src/app/task.ts
export class Task {
  id: number;
  title: string;
  description: string;
  deadline: Date;

  constructor(id:number, title:string, description:string, deadline:Date) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.deadline = deadline;
  }
}
